### PR TITLE
initialize magic6

### PR DESCRIPTION
**Pull request by adrianursu**
http://campion.edu.ro/arhiva/index.php?page=problem&action=view&id=1248

### CODE REVIEW

File: magic6/magic6.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	for (int i = 1; i <= n; i++)
+	{
+		input >> character;
+		verifyIfItsIntegerOrChar();
+	}
+}
+
+int main()
+{
+	readAndWriteLetters();
```
Comment (danielursuu) [inline magic6/magic6.cpp]: it is not ok that a function does multiple actions

File: magic6/magic6.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+void verifyIfItsIntegerOrChar()
+{
+	if (character > 47 && character < 58)
+	{
+		frequencyVector[character - '0'] = 1;
+	}
+	else
+	{
+		character = (char)(character - 32);
+		output << character;
```
Comment (danielursuu) [inline magic6/magic6.cpp]: variable character is overwritten here
you should output directly (char)(character - 32)

File: magic6/magic6.cpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+using namespace std;
+
+ifstream input("magic6.in");
+ofstream output("magic6.out");
+
+char character;
+int n;
+int frequencyVector[10] = {};
+
+void verifyIfItsIntegerOrChar()
```
Comment (danielursuu) [inline magic6/magic6.cpp]: now you have integer or char. What will you do if a new data type would appear in the context? You should change the name, maybe processCharacters(), separateCharacters(), etc

File: magic6/magic6.cpp (hunk 1 above)
Comment (danielursuu) [reply magic6/magic6.cpp]: please change the name of the function, you don't have just letters here
read() seems to be ok for me

File: magic6/magic6.cpp [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	{
+		character = (char)(character - 32);
+		output << character;
+	}
+}
+
+void writeNumbers()
+{
+	int poz = -1;
+	int i = 1;
```
Comment (danielursuu) [inline magic6/magic6.cpp]: move declaration of i after if statemenet from line 31

File: magic6/magic6.cpp [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	}
+	else
+	{
+		for (int i = 1; i < 10; i++)
+		{
+			if (frequencyVector[i] == 1)
+			{
+				output << i;
+			}
+		}
```
Comment (danielursuu) [inline magic6/magic6.cpp]: duplicate code

File: magic6/magic6.cpp [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+		output << poz << 0;
+		for (int i = poz + 1; i < 10; i++)
+		{
+			if (frequencyVector[i] == 1)
+			{
+				output << i;
+			}
+		}
```
Comment (danielursuu) [inline magic6/magic6.cpp]: duplicate code

File: magic6/magic6.cpp [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+			if (frequencyVector[i] == 1)
+			{
+				poz = i;
+			}
+			else
+			{
+				i++;
+			}
+		}
+		output << poz << 0;
```
Comment (danielursuu) [inline magic6/magic6.cpp]: you should wrap up in a function

**Review by danielursuu — CHANGES_REQUESTED** [review]

File: magic6/magic6.cpp (hunk 6 above)
Comment (adrianursu) [reply magic6/magic6.cpp]: here i = poz + 1 but in the previous change request i = 1

File: magic6/magic6.cpp (hunk 7 above)
Comment (adrianursu) [reply magic6/magic6.cpp]: it's not working, if i put it in a function with 2 parameters(one for poz and second for index) and i call it in my writeNumbers() function, output is not showing the correct answer

Comment (adrianursu) [reply magic6/magic6.cpp]: // void verifyFirstDigitAfter0(int position, int index)
// {
// 	while (position < 0)
// 		{
// 			if (frequencyVector[index] == 1)
// 			{
// 				position = index;
// 			}
// 			else
// 			{
// 				index++;
// 			}
// 		}
// 		output << position << 0;
// }

File: magic6/magic6.cpp (hunk 6 above)
Comment (danielursuu) [reply magic6/magic6.cpp]: parameters!

File: magic6/magic6.cpp (hunk 7 above)
Comment (danielursuu) [reply magic6/magic6.cpp]: a parameter which will be modified in the scope should be a referenced one (&)
remember that!!! we discussed about that

Comment (danielursuu) [reply magic6/magic6.cpp]: you don't need the index parameter

Comment (danielursuu) [reply magic6/magic6.cpp]: because index is used only in this scope

File: magic6/magic6.cpp [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+void processNumbers()
+{
+	int poz = -1;
+	int k = 0;
+
+	if (frequencyVector[0] == 1)
+	{
+		verifyFirstDigitAfter0(poz);
```
Comment (danielursuu) [inline magic6/magic6.cpp]: You don t need to pass pos variable
You can return the first position and also change the name getFirstDigitAfterZero()

Comment (danielursuu) [reply magic6/magic6.cpp]: pos=getFirstDigitAfterZero()

File: magic6/magic6.cpp [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+		outputNumbers(poz);
+	}
+	else
```
Comment (danielursuu) [inline magic6/magic6.cpp]: You don t need else statement, because you output the digits in both cases

File: magic6/magic6.cpp [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+		{
+			output << i;
+		}
+	}
+}
+
+void processNumbers()
+{
+	int poz = -1;
+	int k = 0;
```
Comment (danielursuu) [inline magic6/magic6.cpp]: You don t need this variable

**Review by danielursuu — CHANGES_REQUESTED** [review]

**Review by danielursuu — APPROVED** [review]

**Review by danielursuu — APPROVED** [review]

**Pull request merged**